### PR TITLE
Create doc for charmhub page

### PR DESCRIPTION
**Pull request by zxhdaze**
Add link to discourse to create a doc on charmhub page.

### CODE REVIEW

File: charmcraft.yaml [hunk 1]
```diff
@@ -4,6 +4,9 @@ parts:
   charm:
     charm-python-packages: [setuptools < 58]
 
+links:
+  documentation: https://discourse.charmhub.io/t/charm-nginx-doc/15015
+
```
Comment (samuelallan72) [inline charmcraft.yaml]: metadata.yaml exists, so you must put links there.  We can only use the new unified charmcraft.yaml spec if metadata.yaml is not used at all.

```
$ charmcraft -v pack
Starting charmcraft version 2.7.1
Logging execution to '/home/ubuntu/.local/state/charmcraft/log/charmcraft-20240807-083011.743864.log'
Cannot specify 'links' in charmcraft.yaml when 'metadata.yaml' exists
```

**Review by samuelallan72 — CHANGES_REQUESTED** [review]

**Comment by samuelallan72** [comment]
Also, the documentation will need some work; the readme was outdated, and we don't want developer documentation in the charmcraft docs page.

- remove the outdated `cs:` prefix
- remove developing and testing sections
- test that the example config for enabling tls still works

File: charmcraft.yaml (hunk 1 above)
Comment (zxhdaze) [reply charmcraft.yaml]: Removed metadata to mirror https://github.com/canonical/charm-cloudsupport/pull/28/files

File: charmcraft.yaml [hunk 2]
```diff
@@ -22,3 +27,6 @@ bases:
           channel: "18.04"
           architectures:
               - amd64
+requires:
+  publish:
+    interface: "web-publish"
```
Comment (samuelallan72) [inline charmcraft.yaml]: nit: Missing trailing newline

**Review by samuelallan72 — DISMISSED** [review]
Thanks, looks good to me.

**Review by aieri — DISMISSED** [review]

**Review by gabrielcocenza — APPROVED** [review]

**Review by aieri — APPROVED** [review]

**Pull request merged**